.github/workflows/update_golang_version.yml

```diff
@@ -39,8 +39,10 @@ jobs:
 
           if [ ${{ matrix.branch }} == "main" ]; then
             go run ./go/tools/go-upgrade/go-upgrade.go upgrade --main --allow-major-upgrade
-          else
+          else if [ ${{ matrix.branch }} == "release-21.0" ]; then
             go run ./go/tools/go-upgrade/go-upgrade.go upgrade
```

Comment:
We changed how our workflows are written in `v21.0`, meaning that in `>= v21.0` the tool no longer need to update the workflow files, but in `< v21.0` we still need to upgrade them, but we cannot modify CI workflows from a CI-triggered PR, so for all `< v21.0` branches we use the `--workflow-update=false` flag.